tests/test_run.py

```diff
@@ -48,6 +61,274 @@ async def test_run(async_flag, record_mode):
+    router.route(method="POST", path="/predictions", headers={"Prefer": "wait"}).mock(
+        return_value=httpx.Response(
+            201,
+            json=_prediction_with_status(
+                "processing",
+                [
+                    "Hello, ",
+                    "world!",
+                ],
+            ),
+        )
+    )
+    router.route(
+        method="GET",
+        path="/models/test/example/versions/v1",
+    ).mock(
+        return_value=httpx.Response(
+            201,
+            json=_version_with_schema(
+                "p1",
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                    },
+                    "x-cog-array-type": "iterator",
+                },
+            ),
+        )
+    )
+
+    client = Client(
+        api_token="test-token", transport=httpx.MockTransport(router.handler)
+    )
+    client.poll_interval = 0.001
+
+    stream = cast(
+        Iterator[FileOutput],
+        client.run(
+            "test/example:v1",
+            input={
+                "text": "Hello, world!",
+            },
+        ),
+    )
+
+    output1 = next(stream)
+    output2 = next(stream)
+    with pytest.raises(StopIteration):
+        next(stream)
+
+    assert output1 == "Hello, "
+    assert output2 == "world!"
+
+
+@pytest.mark.asyncio
+async def test_async_run_blocking_with_iterator(mock_replicate_api_token):
+    router = respx.Router(base_url="https://api.replicate.com/v1")
+    router.route(method="POST", path="/predictions", headers={"Prefer": "wait"}).mock(
+        return_value=httpx.Response(
+            201,
+            json=_prediction_with_status(
+                "processing",
+                [
+                    "Hello, ",
+                    "world!",
+                ],
+            ),
+        )
+    )
+    router.route(
+        method="GET",
+        path="/models/test/example/versions/v1",
+    ).mock(
+        return_value=httpx.Response(
+            201,
+            json=_version_with_schema(
+                "p1",
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                    },
+                    "x-cog-array-type": "iterator",
+                },
+            ),
+        )
+    )
+
+    client = Client(
+        api_token="test-token", transport=httpx.MockTransport(router.handler)
+    )
+    client.poll_interval = 0.001
+
+    stream = cast(
+        AsyncIterator[FileOutput],
+        await client.async_run(
+            "test/example:v1",
+            input={
+                "text": "Hello, world!",
+            },
+        ),
+    )
+
+    output1 = await anext(stream)
+    output2 = await anext(stream)
+    with pytest.raises(StopAsyncIteration):
+        await anext(stream)
+
+    assert output1 == "Hello, "
+    assert output2 == "world!"
+
+
```

Comment:
I think we probably also want to cover `Prefer: wait` but it "times out" and returns a prediction in status `"starting"`.